front/components/file/FileUploader.vue

```diff
@@ -124,10 +171,60 @@ export default {
       uploads: [],
       uploading: false,
       hostedFiles: [],
+      memberships: [],
+      availableIndices: [],
+      indicesOfRepo: {},
       loading: false,
+      noIndexSymbol,
     };
   },
   computed: {
+    institutionsOfUser() {
+      return new Map(this.memberships.map((m) => [m.institutionId, m.institution]));
+    },
+    repositoryPermsOfUser() {
+      return (this.$auth.user?.memberships ?? [])
```

Comment:
We are refetching memberships, so lets be consistent and use it everywhere

```suggestion
      return (this.memberships ?? [])
```

Reply:
`this.memberships` and `this.$auth.user.memberships` aren't the same.

`this.memberships` are from `/api/profile/memberships` and only includes institutions without ways to include something else.

`this.$auth.user.memberships` are from `/api/profile` and only includes repositories and spaces, without any way to include something else.